x/payments/handler.go

```diff
@@ -175,7 +206,23 @@ func handleMsgCreatePaymentContract(ctx sdk.Context, k Keeper, bk bank.Keeper,
 	// Submit payment contract
 	k.SetPaymentContract(ctx, contract)
 
-	return sdk.Result{}
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeCreatePaymentContract,
+			sdk.NewAttribute(types.AttributeKeyCreatorDid, msg.CreatorDid),
+			sdk.NewAttribute(types.AttributeKeyPaymentTemplateId, msg.PaymentTemplateId),
+			sdk.NewAttribute(types.AttributeKeyPaymentContractId, msg.PaymentContractId),
+			sdk.NewAttribute(types.AttributeKeyPayer, msg.Payer.String()),
+			sdk.NewAttribute(types.AttributeKeyDiscountId, msg.DiscountId.String()),
+			sdk.NewAttribute(types.AttributeKeyDeAuthorise, strconv.FormatBool(msg.CanDeauthorise)),
```

Comment:
Attribute key should match the value we're outputting, so it should be `AttributeKeyCanDeauthorise`

Reply:
resolved the naming of AttributeKeyCanDeauthorise , AttributeKeyAgentRole